File: dblibrary/src/main/java/com/yubin/dblibrary/DatabaseMigration.java
package com.yubin.dblibrary;

import androidx.annotation.NonNull;
import androidx.room.migration.Migration;
import androidx.sqlite.db.SupportSQLiteDatabase;

/**
 * <pre>
 *     @author : dingyubin
 *     e-mail : dingyubin@gmail.com
 *     time    : 2022/02/08
 *     desc    : 版本 执行sql
 *     version : 1.0
 * </pre>
 */
public class DatabaseMigration {
    public static final Migration MIGRATION_1_2 = new Migration(1, 2) {
        @Override
        public void migrate(@NonNull SupportSQLiteDatabase database) {
            database.execSQL("delete from conversation");
            database.execSQL("delete from chat_record");
            database.execSQL("ALTER TABLE chat_record ADD COLUMN msg_seq INTEGER NOT NULL DEFAULT -1");
        }
    };
}

File: kotlindb/src/main/java/com/yubin/kotlindb/DatabaseMigration.kt
package com.yubin.kotlindb

import androidx.room.migration.Migration
import androidx.sqlite.db.SupportSQLiteDatabase

/**
 * <pre>
 * @author : dingyubin
 * e-mail : dingyubin@gmail.com
 * time    : 2022/02/08
 * desc    : 版本 执行sql
 * version : 1.0
</pre> *
 */
object DatabaseMigration {
    val MIGRATION_1_2: Migration = object : Migration(1, 2) {
        override fun migrate(db: SupportSQLiteDatabase) {
//            db.execSQL("delete from conversation")
            db.execSQL("delete from oe_record")
//            db.execSQL("ALTER TABLE oe_record ADD COLUMN msg_seq INTEGER NOT NULL DEFAULT -1")
        }
    }
}